storage/watchable_store.go

```diff
@@ -64,7 +64,7 @@ type cancelFunc func()
 func newWatchableStore(b backend.Backend, le lease.Lessor) *watchableStore {
 	s := &watchableStore{
 		store:    NewStore(b, le),
-		unsynced: make(map[*watcher]struct{}),
+		unsynced: make(map[string]map[*watcher]struct{}),
```

Comment:
this is a complicated type-- `type WatchersByKey map[string]map[*watcher]struct{}`? `KeyedWatchers`? Not sure on a good name